test/upgrade/functions_test.go

```diff
@@ -120,22 +120,26 @@ func TestSuiteExecuteWithComplete(t *testing.T) {
 	if c.T.Failed() {
 		return
 	}
-	txt := expectedTexts(suite, fp)
-	txt.append(upgradeTestRunning, upgradeTestSuccess)
-	txt.append(
+	expected := expectedTexts(suite, fp)
+	expected.append(upgradeTestRunning, upgradeTestSuccess)
+	expected.append(
 		"Installing Serving stable 0.17.1",
 		"Installing Eventing stable 0.17.2",
-		"Running Serving continual test",
-		"Stopping and verify of Eventing continual test",
```

Comment:
Could you explain why those texts are not expected any more? I presume that's connected to the other my comment - logging only on failure.

Reply:
The tests are setting up a logger/buffer at the beginning and passing that to `upgrade.Configuration`. Later they expect the messages to be there in the buffer. But the framework now uses a different logger/buffer for the background operations which are only logged to testing.T when the test fails. So, when the test fails the messages should be on stdout (through testing.T), when the test passes the messages are not printed..
If we want to specifically verify these messages I can pass a separate logger like I did in the TestSkipAtBackgroundVerification

Reply:
Now, the logs are written on the output in successful state as well - after other my comment.

So, with that change, we should see expected logs in the end.

Reply:
They're written to t.Log, not to the original logger which this test is checking. So, one way to test it would be similar to what I did for the skipped test below - pass a separate logger for the purpose of the test.